### ISSUE TITLE
TypeChecker.fromName does not work with built-in dart packages

### ISSUE THREAD

**Issue by oskar-zeinomahmalat-sonarsource** [invertase/dart_custom_lint#278]
**Describe the bug**
The type check utility `_NamedChecker` returned by `TypeChecker.fromName('name', packageName: 'package')` assumes that `packageName` is the name of a package addressable with a `package:` URL. When using the `isExactly(element)` comparison, `element.library.identifier` is compared against `'package:$packageName/'`.

This behavior prohibits the type checking of Types from built-in dart packages, as their identifiers start with `dart:`.

**To Reproduce**

This lint rule should detect `File.open()` invocations. Currently, it does not work as I would expect.

```dart
const _fileChecker = TypeChecker.fromName('File', packageName: 'dart:io');

class FileOpen extends DartLintRule {
  FileOpen() : super(code: _code);

  static const _code = LintCode(
    name: 'file_open_invocation',
    problemMessage: 'File open method invocations are bad!',
    errorSeverity: ErrorSeverity.ERROR,
  );

  @override
  void run(CustomLintResolver resolver, ErrorReporter reporter, CustomLintContext context) {
    context.registry.addMethodInvocation((node) {
      var type = node.realTarget?.staticType;
      if (type == null) return;
      if (!_fileChecker.isExactlyType(type)) return;
      if (node.methodName.name != 'open') return;
      reporter.atNode(node, _code);
    });
  }
}
```

**Expected behavior**
I expect `_NamedChecker.isExactly` to have extra handling for packageNames starting with `dart:`. In these cases, no string modification should take place to compare library identifiers.

I'm open to feedback if I understood something wrong here, it is my first time using custom_lint.

**Comment by rrousselGit** [comment]
It's low priority for me. Feel free to make a pr :)
